### PR TITLE
[NEW] Collect data about LDAP, SAML, CAS and OAuth usage.

### PR DESCRIPTION
**Pull request by pierre-lehnen-rc**

### CODE REVIEW

File: app/models/server/models/Users.js [hunk 1]
```diff
@@ -1560,6 +1560,18 @@ Find users to send a message by email if:
...
 	}
 
+	countActiveUsersByService(serviceName, options) {
+		const query = {
+			active: true,
+			type: { $nin: ['app'] },
+			roles: { $ne: ['guest'] },
+		};
+
+		query[`services.${ serviceName }`] = { $exists: true };
```
Comment (ggazzo) [inline app/models/server/models/Users.js:1570]:
```suggestion
			roles: { $ne: ['guest'] },
			[`services.${ serviceName }`]:  { $exists: true }
		};
```

File: app/statistics/server/lib/getServicesStatistics.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+			enabled: value,
+			mergeRoles: settings.get(`Accounts_OAuth_Custom-${ name }-merge_roles`),
+			users: Users.countActiveUsersByService(name),
+		};
+	});
+
+	return data;
+}
+
+export function getServicesStatistics(): Record<string, any> {
```
Comment (ggazzo) [inline app/statistics/server/lib/getServicesStatistics.ts:20]: provide something more specific than `any`, if you really dont know, use `unknown` but try to provide some kind of structure

**Review by ggazzo — CHANGES_REQUESTED** [review]
almost there ;)

**Review by ggazzo — APPROVED** [review]

File: app/statistics/server/lib/getServicesStatistics.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import { settings } from '../../../settings/server';
+import { Users } from '../../../models/server';
+
+function getCustomOAuthServices(): Record<string, {
+	enabled: boolean;
+	mergeRoles: boolean;
+	users: number;
+}> {
+	const customOatuh = settings.get(/Accounts_OAuth_Custom-[^-]+$/mi);
+	return Object.fromEntries(customOatuh.map(({ key, value }) => [key, {
```
Comment (geekgonecrazy) [inline app/statistics/server/lib/getServicesStatistics.ts:10]: is this not a typo that will prevent from working @pierre-lehnen-rc ?

**Comment by geekgonecrazy** [comment]
At some point we really should consider making that entire payload a type.  It would make it much easier to work with.  Right now to determine what type the value is supposed to be is a really involved exercise

**Pull request merged**